### ISSUE TITLE
Issue: error clinica convert oasis3-to-bids Oasis3test Oasis3test/ClinicalData Oasis3BidsRun 

### ISSUE THREAD

**Issue by SimonChenS** [aramis-lab/clinica#977]
I got error when I do:
clinica convert oasis3-to-bids Oasis3test Oasis_csv Oasis3Bids

2023-09-27 06:06:39,420:INFO:All clinical data have been found
Traceback (most recent call last):
  File "/Users/simon/miniconda3/envs/clinicaEnv/bin/clinica", line 8, in <module>
    sys.exit(main())
             ^^^^^^
  File "/Users/simon/miniconda3/envs/clinicaEnv/lib/python3.11/site-packages/clinica/cmdline.py", line 92, in main
    cli()
  File "/Users/simon/miniconda3/envs/clinicaEnv/lib/python3.11/site-packages/click/core.py", line 1157, in __call__
    return self.main(*args, **kwargs)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/Users/simon/miniconda3/envs/clinicaEnv/lib/python3.11/site-packages/click/core.py", line 1078, in main
    rv = self.invoke(ctx)
         ^^^^^^^^^^^^^^^^
  File "/Users/simon/miniconda3/envs/clinicaEnv/lib/python3.11/site-packages/click/core.py", line 1688, in invoke
    return _process_result(sub_ctx.command.invoke(sub_ctx))
                           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/Users/simon/miniconda3/envs/clinicaEnv/lib/python3.11/site-packages/click/core.py", line 1688, in invoke
    return _process_result(sub_ctx.command.invoke(sub_ctx))
                           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/Users/simon/miniconda3/envs/clinicaEnv/lib/python3.11/site-packages/click/core.py", line 1434, in invoke
    return ctx.invoke(self.callback, **ctx.params)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/Users/simon/miniconda3/envs/clinicaEnv/lib/python3.11/site-packages/click/core.py", line 783, in invoke
    return __callback(*args, **kwargs)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/Users/simon/miniconda3/envs/clinicaEnv/lib/python3.11/site-packages/clinica/iotools/converters/oasis3_to_bids/oasis3_to_bids_cli.py", line 26, in cli
    convert_images(dataset_directory, bids_directory, clinical_data_directory)
  File "/Users/simon/miniconda3/envs/clinicaEnv/lib/python3.11/site-packages/clinica/iotools/converters/oasis3_to_bids/oasis3_to_bids.py", line 32, in convert_images
    imaging_data = read_imaging_data(path_to_dataset)
                   ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/Users/simon/miniconda3/envs/clinicaEnv/lib/python3.11/site-packages/clinica/iotools/converters/oasis3_to_bids/oasis3_utils.py", line 94, in read_imaging_data
    file_spec_series.str.split("_", expand=True),
    ^^^^^^^^^^^^^^^^^^^^
  File "/Users/simon/miniconda3/envs/clinicaEnv/lib/python3.11/site-packages/pandas/core/generic.py", line 5902, in __getattr__
    return object.__getattribute__(self, name)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/Users/simon/miniconda3/envs/clinicaEnv/lib/python3.11/site-packages/pandas/core/accessor.py", line 182, in __get__
    accessor_obj = self._accessor(obj)
                   ^^^^^^^^^^^^^^^^^^^
  File "/Users/simon/miniconda3/envs/clinicaEnv/lib/python3.11/site-packages/pandas/core/strings/accessor.py", line 181, in __init__
    self._inferred_dtype = self._validate(data)
                           ^^^^^^^^^^^^^^^^^^^^
  File "/Users/simon/miniconda3/envs/clinicaEnv/lib/python3.11/site-packages/pandas/core/strings/accessor.py", line 235, in _validate
    raise AttributeError("Can only use .str accessor with string values!")
AttributeError: Can only use .str accessor with string values!. Did you mean: 'std'?

I tried to find the oasis3_to_bids.py at line 94 and add .astype(str) into it to solve the problem, but it raised another problem. Thus I wonder why would I have this error. I downloaded the data and CSV files and followed the instructions on the website step by step. I use clinica, version 0.7.6 and python 3.11 . 

Would you mind helping me with that? I have struggled with it for a while.

**Comment by SimonChenS** [comment]
@NicolasGensollen @ghisvail Dear Engineers, I am sorry to mention you. Would you mind taking a look at it? I would really appreciate it if you could solve my problem.

**Comment by MatthieuJoulot** [comment]
Hi @SimonChenS,

Looking into it, somethings seems strange about your error. You're supposed to have `file_spec_series.str.split("_", expand=True)` at line 94 of oasis_utils.py and in your error you report having `file_spec_series.str.split("", expand=True)`. I do not know if that is your error, but it does look strange.

Also, what OS are you working on ?

**Comment by SimonChenS** [comment]
Thanks for your reply! I checked the oasis3_utils.py  and there is a "_". Here is a copy of it:

    df_source = pd.concat(
        [
            source_path_series,
            file_spec_series,
            source_dir_series,
            file_spec_series.str.split("_", expand=True),
            source_file_series,
            source_run_series,
        ],
        axis=1,
    )

. I am using the Mac terminal to run the code. Thank you again for your response. 
<img width="1440" alt="Screen Shot 2023-09-28 at 2 55 20 PM copy" src="https://github.com/aramis-lab/clinica/assets/131853762/e00db694-dfc3-41e1-b919-6b9c546d32ce">

**Comment by SimonChenS** [comment]
@MatthieuJoulot I forgot to mention you in the previous response :) thank you!

**Comment by SimonChenS** [comment]
I think the Date and templateType in the CSV file are the problems, they are float types and lead to the question. What should I do?

**Comment by ghisvail** [comment]
Hi Simon,

1. How did you download the dataset? Through XNAT central or the OASIS scripts?
2. Could you give us an excerpt of the list of downloaded files? You could use `ls` or `tree` commands and give us the first 10 entries.

I suspect the path format may have changed, explaining why the parsing of `file_spec_series` fails for you.

**Comment by ghisvail** [comment]
I just went through the instructions in Clinica's documentation and tried converting subject `OAS30001` with the most recent clinical data and using the official scripts to download the imaging data. No problem.

**Comment by SimonChenS** [comment]
Dear ghisvali,

1. I tried using XNAT, however, it seems to take a very long time and then failed. Thus I use the zip format and unzip it.
2. 
<img width="1440" alt="Screen Shot 2023-09-29 at 3 42 28 PM" src="https://github.com/aramis-lab/clinica/assets/131853762/ea1db407-db69-4fd8-b318-b6a5581c5f4c">

I initialized another download using XNAT, I hope it will work well. Thank you so much for your reply!
